src/Processors/Formats/Impl/ValuesBlockInputFormat.cpp

```diff
@@ -181,6 +182,35 @@ bool ValuesBlockInputFormat::tryReadValue(IColumn & column, size_t column_idx)
     }
 }
 
+namespace
+{
+    void tryToReplaceNullFieldsInTupleWithDefaultValues(Field & value, const IDataType & type)
```

Comment:
How does it work for arrays of tuples and tuples of arrays of tuples, etc...?

Reply:
Updated for Array.